TCAT/Controllers/HomeMapViewController.swift

```diff
@@ -0,0 +1,251 @@
+        navigationController?.isNavigationBarHidden = false
+        reachability?.stopNotifier()
+        
+        // Remove Notification Observers
+        NotificationCenter.default.removeObserver(self, name: .reachabilityChanged, object: reachability)
+        
+        // Remove banner and loading indicator
+        banner?.dismiss()
+        banner = nil
+    }
+    
+    func setupMapView() {
+        // Set mapView with settings
+        let camera = GMSCameraPosition.camera(withLatitude: 42.446179, longitude: -76.485070, zoom: defaultZoom)
+        let mapView = GMSMapView.map(withFrame: .zero, camera: camera)
+        mapView.delegate = self
+        mapView.isMyLocationEnabled = true
+        mapView.paddingAdjustmentBehavior = .never // handled by code
+        mapView.setMinZoom(minZoom, maxZoom: maxZoom)
+        mapView.settings.compassButton = true
+        mapView.settings.myLocationButton = true
+        mapView.settings.tiltGestures = false
+        mapView.settings.indoorPicker = false
+        mapView.isBuildingsEnabled = false
+        mapView.isIndoorEnabled = false
+        mapView.padding = .init(top: 0, left: 0, bottom: 10, right: 10)
+
+        let northEast = CLLocationCoordinate2D(latitude: Constants.Values.RouteMaxima.north, longitude: Constants.Values.RouteMaxima.east)
+        let southWest = CLLocationCoordinate2D(latitude: Constants.Values.RouteMaxima.south, longitude: Constants.Values.RouteMaxima.west)
+        let panBounds = GMSCoordinateBounds(coordinate: northEast, coordinate: southWest)
+        mapView.cameraTargetBounds = panBounds
+        
+        self.mapView = mapView
+        view.addSubview(mapView)
+        mapView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+    
+    func setupOptionsCard() {
+        optionsCardVC = HomeOptionsCardViewController()
+        add(optionsCardVC)
+        optionsCardVC.delegate = self
+        delegate = optionsCardVC
+    }
+    
+    func setupConstraints() {
+        optionsCardVC.view.snp.makeConstraints { (make) in
+            make.leading.top.trailing.equalToSuperview().inset(HomeMapViewController.optionsCardInset)
+            make.height.equalTo(optionsCardVC.calculateCardHeight())
+        }
+    }
+    
+    /// Show a temporary loading screen
+    func showLoadingScreen() {
+        
+        loadingView.backgroundColor = Colors.backgroundWash
+        view.addSubview(loadingView)
+        
+        loadingView.snp.makeConstraints { (make) in
+            make.top.bottom.leading.trailing.equalToSuperview()
+        }
+        
+        let indicator = LoadingIndicator()
+        loadingView.addSubview(indicator)
+        indicator.snp.makeConstraints { (make) in
+            make.center.equalToSuperview()
+            make.size.equalTo(loadingIndicatorSize)
+        }
+        
+    }
+    
+    func removeLoadingScreen() {
+        loadingView.removeFromSuperview()
+        viewWillAppear(false)
```

Comment:
we shouldn't call `viewWillAppear` directly. if theres code that needs to be reused, its prob better to extract that to a helper function and call that instead